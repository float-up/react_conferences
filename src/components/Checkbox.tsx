import React from "react";

interface IProps {
    text: string;
    type?: string;
    checked?: boolean;
    onChange: (text: string) => void;
}


export const Checkbox: React.FC<IProps> = props => {

    const {
        text,
        onChange,
        checked
    } = props;

    return (
        <label className="checkbox">
            <input id="indeterminate-checkbox" type="checkbox" checked={checked} onChange={() => onChange(text)} />
            <span>{text}</span>
        </label>
    )
}


