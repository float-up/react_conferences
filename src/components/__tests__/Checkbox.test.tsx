import React from "react";
import {shallow} from "enzyme";
import {Checkbox} from "../Checkbox";

const handleClick = jest.fn();
let wrapper = shallow(<Checkbox text="Ru" onChange={handleClick}/>)
describe('Checkbox', () => {
    it('test Checkbox render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('test props "text" in Checkbox', () => {
        expect(wrapper.find('span').text()).toEqual('Ru')
    })

    it('toggle Checkbox', () => {

        wrapper.find('input')
            .simulate('change', {target: {checked: true}})

        expect(handleClick).toHaveBeenCalled();
    })
})
