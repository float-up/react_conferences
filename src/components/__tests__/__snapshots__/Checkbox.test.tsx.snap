// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox test Checkbox render 1`] = `
<label
  className="checkbox"
>
  <input
    id="indeterminate-checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span>
    Ru
  </span>
</label>
`;
